### PR TITLE
graphene-hardened-malloc: 8 -> 11

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

- version update:
  * https://github.com/GrapheneOS/hardened_malloc/releases/tag/11
  * https://github.com/GrapheneOS/hardened_malloc/releases/tag/10
  * https://github.com/GrapheneOS/hardened_malloc/releases/tag/9

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg test hardened

**Comment by risicle** [comment]
Passthru tests aren't passing.

**Comment by prusnak** [comment]
> Passthru tests aren't passing.

The error seems to be `stress-ng: error: [5] no CPUs found in /sys/devices/system/cpu`

Is that failure relevant to this PR?

**Comment by risicle** [comment]
No that's not the error. Much higher up:

```
/nix/store/xcy53qbjynin70g46w1nx011gr7jw8l1-stdenv-linux/setup: line 1352: pushd: test/simple-memory-corruption: No such file or directory
builder for '/nix/store/jdlv610jkjqx6p4lnlgzrrg36hdgrbk0-graphene-hardened-malloc-11-ld-preload-tests.drv' failed with exit code 1
```

**Comment by risicle** [comment]
Have they moved the tests?

**Comment by prusnak** [comment]
> Have they moved the tests?

Right. Yes, they did. Updated the PR in babba8a01a8b48698330f54a398dd1c4c71d3c9f

**Comment by prusnak** [comment]
@ofborg test hardened

**Comment by risicle** [comment]
`nixosTests.hardened` appears already broken for other reasons :( #151696

**Comment by risicle** [comment]
I'm still getting

```
/nix/store/xcy53qbjynin70g46w1nx011gr7jw8l1-stdenv-linux/setup: eval: line 1357: unexpected EOF while looking for matching `"'
```

for `graphene-hardened-malloc.tests`

**Comment by risicle** [comment]
@ofborg build graphene-hardened-malloc.tests

**Comment by risicle** [comment]
Yup ofborg still getting the same failure:

```
/nix/store/h6bva8mkj6xfp7za4v3jpp59i58wqs9f-stdenv-linux/setup: eval: line 1359: unexpected EOF while looking for matching `"'
builder for '/nix/store/ylzxfmfq6n8ng7wrdnajnddaa7n1r2pn-graphene-hardened-malloc-11-ld-preload-tests.drv' failed with exit code 2
```

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 157413` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>graphene-hardened-malloc</li>
  </ul>
</details>

**Comment by risicle** [comment]
@QuantMint could you try building `graphene-hardened-malloc.tests` with that? Expect it will fail.

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 157413` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>graphene-hardened-malloc</li>
  </ul>
</details>

**Comment by risicle** [comment]
I'm not sure anything was any different with that.

Try `nixpkgs-review pr 157413 -p graphene-hardened-malloc.tests.ld-preload`

**Comment by QuantMint** [comment]
> 

It doesn't seem to work:
 `The following packages specified with -p are not rebuilt by the pull request
graphene-hardened-malloc.tests.ld-preload`

**Comment by risicle** [comment]
@ofborg build graphene-hardened-malloc.tests

File: pkgs/development/libraries/graphene-hardened-malloc/default.nix [hunk 1]
```diff
@@ -41,21 +48,21 @@ lib.fix (self: stdenv.mkDerivation rec {
...
 
       installPhase = ''
         mkdir -p $out/test
-        cp -r test/simple-memory-corruption $out/test/simple-memory-corruption
+        cp -r test $out/test
 
         mkdir -p $out/bin
         makeWrapper ${python3.interpreter} $out/bin/run-tests \
-          --add-flags "-I -m unittest discover --start-directory $out/test/simple-memory-corruption"
+          --add-flags "-I -m unittest discover --start-directory $out/test
```
Comment (risicle) [inline pkgs/development/libraries/graphene-hardened-malloc/default.nix:65]: Spotted the problem. Missing close quote.

**Review by risicle — APPROVED** [review]

**Pull request merged**